pkg/restic/exec_commands.go

```diff
@@ -53,3 +62,91 @@ func GetSnapshotID(repoIdentifier, passwordFile string, tags map[string]string,
 
 	return snapshots[0].ShortID, nil
 }
+
+// RunBackup runs a `restic backup` command and watches the output to provide
+// progress updates to the caller.
+func RunBackup(backupCmd *Command, log logrus.FieldLogger, updateFunc func(velerov1api.PodVolumeOperationProgress)) (string, string, error) {
```

Comment:
One issue with this is that if a backup completes too quickly, the progress will never get set. This can easily happen for smaller volumes or if only a few files have changed in an incremental backup. I'm now thinking that this function should call `updateFunc` one last time with the final progress before returning, and we don't need to set the progress to 100% in the controller.